File: SonicXBackned/src/main/java/org/example/PlaylistManagment/Playlist.java
package org.example.PlaylistManagment;

import org.example.MusicManagment.MusicFile;
import org.example.UserManagment.UserService;

import java.sql.SQLException;
import java.util.List;

public class Playlist {
    private String name;
    private PlaylistDAO playlistDAO;

    public Playlist(String name) {
        this.name = name;
        try {
            this.playlistDAO = new PlaylistDAO();
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public String getName() {
        return name;
    }

    public void addMusicFileToPlaylist(MusicFile musicFile, UserService userService) {
        try {
            playlistDAO.addMusicFileToPlaylist(this.name, musicFile,userService);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public void removeMusicFileFromPlaylist(String musicTitle,UserService userService) {
        try {
            playlistDAO.removeMusicFileFromPlaylist(this.name, musicTitle,userService);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public List<MusicFile> getAllMusicFiles(UserService userService) {
        try {
            return playlistDAO.getMusicFilesInPlaylist(this.name,userService);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public void close() {
        playlistDAO.close();
    }

}

File: SonicXBackendC/SonicXBackendC/PlaylistManagment/Playlist.cs
using System;
using System.Collections.Generic;
using SonicXBackendC.MusicManagment;
using SonicXBackendC.UserManagement;

namespace SonicXBackendC.PlaylistManagment
{
    public class Playlist
    {
        private string name;
        private PlaylistDAO playlistDAO;

        public Playlist(string name)
        {
            this.name = name;
            try
            {
                this.playlistDAO = new PlaylistDAO();
            }
            catch (Exception e)
            {
                throw new Exception("Veritabanı bağlantısı oluşturulurken hata oluştu: " + e.Message);
            }
        }

        public string Name
        {
            get { return name; }
        }

        public void AddMusicFileToPlaylist(MusicFile musicFile, UserService userService)
        {
            try
            {
                playlistDAO.AddMusicFileToPlaylist(this.name, musicFile, userService);
            }
            catch (Exception e)
            {
                throw new Exception("Müzik dosyası eklenirken hata oluştu: " + e.Message);
            }
        }

        public void RemoveMusicFileFromPlaylist(string musicTitle, UserService userService)
        {
            try
            {
                playlistDAO.RemoveMusicFileFromPlaylist(this.name, musicTitle, userService);
            }
            catch (Exception e)
            {
                throw new Exception("Müzik dosyası silinirken hata oluştu: " + e.Message);
            }
        }

        public List<MusicFile> GetAllMusicFiles(UserService userService)
        {
            try
            {
                return playlistDAO.GetMusicFilesInPlaylist(this.name, userService);
            }
            catch (Exception e)
            {
                throw new Exception("Müzik dosyaları getirilirken hata oluştu: " + e.Message);
            }
        }

        public void Close()
        {
            playlistDAO.Close();
        }
    }
}
